core/src/main/scala/caliban/parsing/Parser.scala

```diff
@@ -266,6 +268,62 @@ object Parser {
         )
     }
 
+  private def schemaExtension[_: P]: P[SchemaExtension] =
+    P("extend schema" ~/ directives.? ~ "{" ~ rootOperationTypeDefinition.rep ~ "}").map {
+      case (directives, ops) =>
+        val opsMap = ops.toMap
+        SchemaExtension(
+          directives.getOrElse(Nil),
+          opsMap.get(OperationType.Query).map(_.name),
+          opsMap.get(OperationType.Mutation).map(_.name),
+          opsMap.get(OperationType.Subscription).map(_.name)
+        )
+    }
+
+  private def scalarTypeExtension[_: P]: P[ScalarTypeExtension] =
+    P("extend scalar" ~/ name ~ directives).map {
+      case (name, directives) =>
+        ScalarTypeExtension(name, directives)
+    }
+
+  private def objectTypeExtension[_: P]: P[ObjectTypeExtension] =
+    P("extend type" ~/ name ~ implements.? ~ directives.? ~ "{" ~ fieldDefinition.rep ~ "}").map {
```

Comment:
Here too, there could be a case where we provide no fields and only implements/directives: http://spec.graphql.org/June2018/#sec-Object-Extensions

Same for the others below